src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -272,10 +277,10 @@ string EncodeUrl (Uri url)
 				if(Logger.LogNet)
 					Logger.Log (LogLevel.Info, LOG_APP, " cancelled");
 
-				httpConnection.Disconnect ();
+				await DisconnectAsync (httpConnection);
 				cancellationToken.ThrowIfCancellationRequested ();
 			}
-			cancellationToken.Register (httpConnection.Disconnect);
+			cancellationToken.Register (async () => await DisconnectAsync (httpConnection));
```

Comment:
Also it's worth pointing out that if DisconnectAsync throws an exception it will be unobserved. We may want to wrap in a try/catch instead of letting that propagate